app/assets/javascripts/initializers/runtime.js

```diff
@@ -7,6 +7,62 @@
  * if (Runtime.isNativeAndroid('podcastMessage')) { ... }
  */
 class Runtime {
+  /**
+   * This function returns a string combining the current Medium and OS
+   * that represents the current Context where the app is running.
+   *
+   * @returns {String} "Medium-OS", for example "Browser-Android"
+   */
+  static currentContext() {
+    return `${Runtime.currentMedium()}-${Runtime.currentOS()}`;
+  }
+
+  /**
+   * This function returns a string that represents the current Medium where
+   * the app is currently running. The currently supported mediums are Browser,
+   * ForemWebView and PWA.
+   *
+   * @returns {String} One of the supported Mediums or 'Unsupported'
+   */
+  static currentMedium() {
+    const pwaButtons = document.getElementById('pwa-nav-buttons');
+    if (/ForemWebView/i.test(navigator.userAgent)) {
+      return 'ForemWebView';
+    } else if (pwaButtons.classList.contains('pwa-nav-buttons--showing')) {
+      return 'PWA';
+    } else {
+      return 'Browser';
+    }
+  }
+
+  /**
+   * This function returns a string that represents the current OS where the app
+   * is currently running. The currently supported Operating Systems are
+   * Windows, Linux, macOS, Android and iOS.
+   *
+   * @returns {String} One of the supported Operating Systems or 'Unsupported'
+   */
+  static currentOS() {
+    const macosPlatforms = ['Macintosh', 'MacIntel', 'MacPPC', 'Mac68K'];
+    const windowsPlatforms = ['Win32', 'Win64', 'Windows', 'WinCE'];
+    const iosPlatforms = ['iPhone', 'iPad', 'iPod'];
+    let os = 'Unsupported';
+
+    if (macosPlatforms.includes(window.navigator.platform)) {
+      os = 'macOS';
+    } else if (iosPlatforms.includes(window.navigator.platform)) {
+      os = 'iOS';
+    } else if (windowsPlatforms.includes(window.navigator.platform)) {
+      os = 'Windows';
+    } else if (/Android/i.test(window.navigator.userAgent)) {
+      os = 'Android';
+    } else if (/Linux/i.test(window.navigator.platform)) {
+      os = 'Linux';
+    }
+
+    return os;
```

Comment:
suggestion (not-blocking) - instead of reassigning the `os` variable we could just return inside the if statement -
```suggestion

    if (macosPlatforms.includes(window.navigator.platform)) {
      return 'macOS';
    } else if (iosPlatforms.includes(window.navigator.platform)) {
      return 'iOS';
    } else if (windowsPlatforms.includes(window.navigator.platform)) {
      return 'Windows';
    } else if (/Android/i.test(window.navigator.userAgent)) {
      return 'Android';
    } else if (/Linux/i.test(window.navigator.platform)) {
     return 'Linux';
    }

    return 'Unsupported';
```